File: src/api/langboard/ai/LangflowHelper.py
from abc import ABC, abstractmethod
from enum import Enum
from typing import Any, Literal
from core.utils.EditorContentParser import DATA_TEXT_FORMAT_DESCRIPTIONS
from pydantic import BaseModel


class LangflowConstants(Enum):
    ApiKey = "x-api-key"


class LangflowComponent(ABC, BaseModel):
    @abstractmethod
    def to_tweaks(self) -> dict[str, Any]: ...


class LangboardCalledVariablesComponent(LangflowComponent):
    event: str
    app_api_token: str
    project_uid: str | None = None
    bot_labels_for_project: list[dict[str, Any]] = []
    current_runner_type: Literal["bot", "user"]
    current_runner_data: dict[str, Any] | None = None
    rest_data: dict[str, Any] | None = None
    custom_markdown_formats: dict[str, str] = DATA_TEXT_FORMAT_DESCRIPTIONS

    def to_tweaks(self) -> dict[str, Any]:
        return {LangboardCalledVariablesComponent.__name__: self.to_data()}

    def to_data(self) -> dict[str, Any]:
        return self.model_dump()

File: src/socket/src/core/ai/helpers/LangflowHelper.ts
/* eslint-disable @typescript-eslint/no-explicit-any */
import { API_URL } from "@/Constants";
import { DATA_TEXT_FORMAT_DESCRIPTIONS } from "@/core/utils/EditorUtils";

export enum ELangflowConstants {
    ApiKey = "x-api-key",
}

abstract class BaseLangflowComponent {
    public abstract toTweaks(): Record<string, any>;
    public abstract toData(): Record<string, any>;
}

export class LangboardCalledVariablesComponent extends BaseLangflowComponent {
    #model: Record<string, any>;

    constructor(
        event: string,
        app_api_token: string,
        current_runner_type: "bot" | "user",
        current_runner_data?: Record<string, any>,
        project_uid?: string,
        bot_labels_for_project: Array<Record<string, any>> = [],
        rest_data?: Record<string, any>,
        custom_markdown_formats: Record<string, string> = DATA_TEXT_FORMAT_DESCRIPTIONS
    ) {
        super();
        this.#model = {
            base_url: API_URL,
            event,
            app_api_token,
            current_runner_type,
            current_runner_data,
            project_uid,
            bot_labels_for_project,
            rest_data,
            custom_markdown_formats,
        };
    }

    public toTweaks(): Record<string, any> {
        return { LangboardCalledVariablesComponent: this.toData() };
    }

    public toData(): Record<string, any> {
        return { ...this.#model };
    }
}
